src/lib/utils.js

```diff
@@ -208,6 +212,63 @@ const purgeDefaultKeyValueStore = async (cwd) => {
     await Promise.all(deletePromises);
 };
 
+/**
+ * It waits until task is in terminal status(not running).
+ * @param apifyClient
+ * @param task
+ * @param type
+ * @param timeout
+ * @return {Promise<*>}
+ */
+const waitForTaskFinish = async (apifyClient, task, type, timeout) => {
+    const { actId } = task;
+    let taskDetailPromise; let taskOptions;
+    if (type === ACT_TASK_TYPES.BUILD) {
+        taskDetailPromise = apifyClient.acts.getBuild;
+        taskOptions = { actId, buildId: task.id };
+    } else {
+        taskDetailPromise = apifyClient.acts.getRun;
+        taskOptions = { actId, runId: task.id };
+    }
+
+    const startedAt = Date.now();
+    let isRunning = true;
+    let taskDetail;
+    while (isRunning) {
+        taskDetail = await taskDetailPromise(taskOptions); // eslint-disable-line no-await-in-loop
+        if (ACT_TASK_TERMINAL_STATUSES.includes(taskDetail.status) || (timeout && Date.now() - startedAt > timeout)) {
+            isRunning = false;
+        } else {
+            await delayPromise(WAIT_FOR_TASK_INTERVAL); // eslint-disable-line no-await-in-loop
+        }
+    }
+
+    return taskDetail;
+};
+
+const watchStreamLog = (logId, timeout) => {
+    return new Promise((resolve, reject) => {
+        const req = https.get(`https://api.apify.com/v2/logs/${logId}?stream=1`);
+
+        req.on('response', (response) => {
+            response.on('data', chunk => console.log(chunk.toString().trim()));
```

Comment:
I am not 100% sure but you may need to listen also here for error `response.on('error', ...`